lib/screens/wallet/model/crypto_model.dart

```diff
@@ -0,0 +1,46 @@
+
+import 'dart:convert';
+
+import 'package:decimal/decimal.dart';
+
+class CryptoModel {
+  String shortName;
+  String fullName;
+  String cryptoIcon;
+  Decimal currencyCustomerValue;
+  double amountCurrency;
+
+  CryptoModel({
+    required this.shortName,
+    required this.fullName,
+    required this.cryptoIcon,
+    required this.currencyCustomerValue,
+    required this.amountCurrency,
+  });
+
+  Map<String, dynamic> toMap() {
+    final result = <String, dynamic>{};
+  
+    result.addAll({'shortName': shortName});
+    result.addAll({'fullName': fullName});
+    result.addAll({'cryptoIcon': cryptoIcon});
+    result.addAll({'currencyCustomerValue': currencyCustomerValue});
+    result.addAll({'amountCurrency': amountCurrency});
+  
+    return result;
```

Comment:
tu pode fazer direto como... assim n precisa gastar memoria adicionando o item dentro do objeto
```suggestion
    final result = <String, dynamic>{{'shortName': shortName}};
  
    return result;
```